src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -83,6 +83,8 @@ class LicenceService(
       ?: throw RuntimeException("Staff with username $username not found")
 
     val licence = transform(request)
+
+    licence.dateCreated = LocalDateTime.now()
```

Comment:
Just wondering why not do this in the constructor? Then the property is immutable...

Reply:
Its a good question - could be done in the transform(request) - but there are other things which the API enriches after creating the licence object so for simplicity I added to the same section to do them all together.